### PR TITLE
don't link to deprecated function

### PR DESCRIPTION
**Pull request by MichaelChirico**
Seems misleading to (without qualification) link to a deprecated function

## What changes were proposed in this pull request?

(Please fill in changes proposed in this fix)

## How was this patch tested?

(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)
(If this patch involves UI changes, please attach a screenshot; otherwise, remove this)

Please review http://spark.apache.org/contributing.html before opening a pull request.

### CODE REVIEW

**Review by felixcheung — COMMENTED** [review]
thanks for the your input. I'm not sure if this is a big problem - I think it's helpful to link from the deprecated method to the replacement, and back?

File: R/pkg/R/catalog.R [hunk 1]
```diff
@@ -69,7 +69,6 @@ createExternalTable <- function(x, ...) {
 #' @param ... additional named parameters as options for the data source.
 #' @return A SparkDataFrame.
 #' @rdname createTable
-#' @seealso \link{createExternalTable}
```
Comment (HyukjinKwon) [inline R/pkg/R/catalog.R]: One thing we should check for such things is, if it happens globally in the documentation or not BTW.

**Comment by MichaelChirico** [comment]
@felixcheung I disagree... what's the point of deprecation if it's going to keep being considered as a co-equal function in the eyes of documentation?

If the function is being deprecated, it should be "soft-blacklisted" -- documentation still available (but clearly marked) for users of "legacy" code where its usage still survives, but I don't see any argument for even letting new users find out a deprecated function exists.

File: R/pkg/R/catalog.R (hunk 1 above)
Comment (MichaelChirico) [reply R/pkg/R/catalog.R]: I don't see it here (nothing pointing to `sparkR.init`/`sparkRHive.init`/`sparkRSQL.init`):

https://spark.apache.org/docs/latest/api/R/sparkR.session.html

or here (nothing pointing to `dropTempTable`):

https://spark.apache.org/docs/latest/api/R/dropTempView.html

But I do see it here (points to `registerTempTable`):

https://spark.apache.org/docs/latest/api/R/createOrReplaceTempView.html

Comment (MichaelChirico) [reply R/pkg/R/catalog.R]: Shall I amend the PR to fix the `registerTempTable` page as well?

Comment (HyukjinKwon) [reply R/pkg/R/catalog.R]: If we should fix, yea, better fix them all. @felixcheung has a better insight in R side. You batter wait for his opinion.

**Comment by felixcheung** [comment]
I don’t feel strongly either way.

I do think this is very minor since there are still many other ways to the doc page for createExternalTable (eg the index page) or via ? search within R etc. I am not sure how much difference this would make and we have already a) code spewing out warning when called b) clearly documented as Deprecated on the doc page title.

Should you find other deprecation that is not documentation we should be gladly having your help
to document it.

**Review by felixcheung — COMMENTED** [review]
hey thanks for the contribution, I'm good if you could remove the `@family` in `registerTempTable` and check the doc generated by doc build?

**Comment by MichaelChirico** [comment]
@felixcheung thanks for the feedback, yes, I axed `registerTempTable` from the `SparkDataFrame `@family`. 

Sorry I botched an attempt to try and rename the branch of this PR (apparently not possible to overwrite this PR with a new branch name); PR re-filed as #22393

**Pull request closed without merging**